task_sdk/tests/execution_time/test_supervisor.py

```diff
@@ -884,6 +884,18 @@ def watched_subprocess(self, mocker):
                 "",
                 id="patch_task_instance_to_skipped",
             ),
+            # testing to see if supervisor can handle TaskState message with state as fail_with_retry
+            pytest.param(
+                TaskState(
+                    state=TerminalTIState.FAIL_WITHOUT_RETRY,
+                    end_date=timezone.parse("2024-10-31T12:00:00Z"),
+                ),
+                b"",
+                "",
+                (),
+                "",
+                id="patch_task_instance_to_failed_with_retries",
+            ),
```

Comment:
imo this test does not add anything -- and test anything new.

We already check `TaskState` as part of SKIPPED, and since we don't need to assert all the states that are supported by `TaskState`

Reply:
We should remove it

Reply:
Yeah, we do not need it, the message type is enough, this was a carry over from when i called it "FailState". Will remove